src/sprout/Helpers/CustomHeadTags.php

```diff
@@ -0,0 +1,421 @@
+ *
+ * This file is a part of SproutCMS.
+ *
+ * SproutCMS is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU General Public License as published by the Free Software Foundation, either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * For more information, visit <http://getsproutcms.com>.
+ */
+
+namespace Sprout\Helpers;
+
+use Kohana;
+
+/**
+ * Management of custom HEAD tags. Often meta but can also be link, base, etc
+ */
+class CustomHeadTags
+{
+
+    /**
+     * A list of possible tag combinations in the format of tag_type => [tags] => [values]
+     *
+     * NOTE: This can be customised by overriding the same setup in Kohana::config('sprout.custom_meta.available_tag_list')
+     *
+     * @var array
+     */
+    protected static $available_tag_list = [
+        'meta' => [
+            'name' => [
+                'application-name' => [
+                    'content' => 'text',
+                ],
+                'author' => [
+                    'content',
+                ],
+                // 'description' => [
+                //     'content',
+                // ],
+                'generator' => [
+                    'content',
+                ],
+                'keywords' => [
+                    'content',
+                ],
+                'viewport' => [
+                    'content',
+                ],
+                'robots' => [
+                    'content',
+                ],
+                'googlebot' => [
+                    'content',
+                ],
+            ],
+        ],
+        'link' => [
+            'rel' => [
+                'alternate' => [
+                    'type',
+                    'title',
+                    'href',
+                    'hreflang',
+                    'media',
+                ],
+                'canonical' => [
+                    'href',
+                ],
+                // 'icon' => [
+                //     'href',
+                //     'sizes',
+                // ],
+                'stylesheet' => [
+                    'href',
+                    'type',
+                    'media',
+                ],
+                'prev' => [
+                    'href',
+                ],
+                'next' => [
+                    'href',
+                ],
+                'amphtml' => [
+                    'href',
+                ],
+
+            ],
+        ],
+        // Note that for scripts, the "content" is rendered as the script itself, not inside a "content" attribute
+        'script' => [
+            'type' => [
+                'application/ld+json' => [
+                    'content',
+                ],
+            ],
+        ],
+    ];
+
+
+    /**
+     * Get the list of available tags, allowing override form config if set
+     *
+     * @return array
+     */
+    public static function getAvailableTagList()
```

Comment:
We can add strict types here, which should save our butts in future when we inevitably install an invalid config.

```public static function getAvailableTagList(): array```

Same for all the methods in this helper. With a minimum php 7.4 we can also use nullish and void types.

Reply:
Good thinking, doing this now